src/gui/guiFormSpecMenu.cpp

```diff
@@ -3656,18 +3656,13 @@ void GUIFormSpecMenu::drawMenu()
 			NULL, m_client, IT_ROT_HOVERED);
 	}
 
-	// On touchscreens, m_pointer is set by GUIModalMenu::preprocessEvent instead.
-#ifndef HAVE_TOUCHSCREENGUI
-	m_pointer = RenderingEngine::get_raw_device()->getCursorControl()->getPosition();
-#endif
-
 	/*
 		Draw fields/buttons tooltips and update the mouse cursor
 	*/
 	gui::IGUIElement *hovered =
 			Environment->getRootGUIElement()->getElementFromPoint(m_pointer);
 
-#ifndef HAVE_TOUCHSCREENGUI
+#ifndef __ANDROID__
```

Comment:
doesn't have to happen in this PR but I'm sure a better solution can be found here.
e.g. after a quick look `getCursorControl()` just return NULL on android, we should check for that.

Reply:
BTW, we have both `#ifndef __ANDROID__` (common) and `#ifndef ANDROID` (rare). Which one is right?

Reply:
`__ANDROID__` is better. `ANDROID` is also defined (not sure where) but I'd avoid it

Reply:
> e.g. after a quick look `getCursorControl()` just return NULL on android, we should check for that.

Done.